### PR TITLE
Use docker-compose in Github CI so tests are run in the same env

### PR DESCRIPTION
**Pull request by kurdybacha**
* Use docker-compose in Github CI so tests are run in the same env
* Fix web docker and use node 14 as this is official Vercer runtime

### CODE REVIEW

**Review by tomknig — COMMENTED** [review]
Looks good in general. I like the idea of actually building inside docker instead of building locally and copying over build artifacts. Should we maybe add `node_modules` to the `.dockerignore` file so we don't copy then into the image? I think this could otherwise lead to issues on some platforms.

File: Dockerfile [hunk 1]
```diff
@@ -1,16 +1,13 @@
...
 
+USER node
 ENV MONGO_URL "mongodb://mongo:27017/race"
-ENV DB_NAME points
-ENV COL_NAME dataPoints
-
-RUN yarn
 
-COPY . .
+RUN yarn install --frozen-lockfile
```
[the remark this answers is not in the document]
Comment (kurdybacha) [reply Dockerfile:10]: Yeah, we probably need to do that.

File: .github/workflows/yarn-workflow.yml [hunk 2]
```diff
@@ -6,12 +6,9 @@ jobs:
...
-      - uses: borales/actions-yarn@v2.3.0
-        with:
-          cmd: lint # will run `yarn lint` command
-      - uses: borales/actions-yarn@v2.3.0
-        with:
-          cmd: build # will run `yarn build` command
+      - name: Build the stack
+        run: |
+          docker-compose up -d
+          docker container ls
```
[the remark this answers is not in the document]
Comment (kurdybacha) [reply .github/workflows/yarn-workflow.yml:12]: I was thinking we are going to run some tests soon

File: Dockerfile [hunk 3]
```diff
@@ -1,16 +1,13 @@
...
+RUN yarn build
 
-CMD ["yarn", "dev"]
+CMD [ "yarn", "start" ]
```
[the remark this answers is not in the document]
Comment (kurdybacha) [reply Dockerfile:13]: Ack

**Comment by kurdybacha** [comment]
> Looks good in general. I like the idea of actually building inside docker instead of building locally and copying over build artifacts. Should we maybe add `node_modules` to the `.dockerignore` file so we don't copy then into the image? I think this could otherwise lead to issues on some platforms.

**Comment by kurdybacha** [comment]
> Looks good in general. I like the idea of actually building inside docker instead of building locally and copying over build artifacts. Should we maybe add `node_modules` to the `.dockerignore` file so we don't copy then into the image? I think this could otherwise lead to issues on some platforms.

Ack, thanks!

**Pull request merged**